ethstorage/storage_manager.go

```diff
@@ -293,6 +321,164 @@ func (s *StorageManager) syncCheck(kvIdx uint64) error {
 	return nil
 }
 
+// DownloadAllMetas This function download the blob hashes of all the local storage shards from the smart contract
+func (s *StorageManager) DownloadAllMetas() error {
+	header, err := s.l1Source.HeaderByNumber(context.Background(), big.NewInt(rpc.FinalizedBlockNumber.Int64()))
+	if err != nil {
+		return err
+	}
+	l1 := header.Number.Int64()
+
+	lastKvIdx, err := s.l1Source.GetStorageLastBlobIdx(l1)
```

Comment:
I believe it is safer to retrieve lastKvIdx and l1 from local variables with lock.